src/actions/actions.ts

```diff
@@ -1440,6 +1440,24 @@ class CommandInsertNewLineBefore extends BaseCommand {
 }
 
 @RegisterAction
+class CommandClearLine extends BaseCommand {
+  modes = [ModeName.Normal];
+  keys = ["S"];
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const lineStart = position.getFirstLineNonBlankChar();
```

Comment:
@johnfn That wasn't my understanding from #536, or from my usage of vim.

For example, consider a sample function:

```
 def func(one, two):
     print o|ne
     print two
```

By pressing `S` I would expect to replace the second line with the correct indent, which this accomplishes.

Just nuking the whole line seems less useful, and not what my terminal vim does.

I'm happy to change to first char, but is that really desired?